### PR TITLE
25 Create role service

### LINKED ISSUE #25: Create RoleService
Create RoleService which has three methods:

- addRole which takes RoleRequestDTO and adds Role. It should return new Role Object
- getRole which takes role id and returns Optional of Role
- getRoles which takes pageable and returns Page of Roles

Controller method addRole it should be void and return location header.
(Resource https://shivaprasadgurram.hashnode.dev/how-to-add-location-header-to-spring-mvcs-post-response)
Attach given methods to respective methods in controller and make some tests to test happy flow of application.
Test scenarios for controller tests:
- Adding valid role should create role successfully
- Retrieving existing role should return should
- Retrieving not existing role by id should return 404 status code
- It should return Return role in given range. For example from second to fourth.
Links 
https://www.baeldung.com/spring-boot-testing
https://www.baeldung.com/spring-boot-testcontainers-integration-test

### PR DESCRIPTION
**Pull request by Boomber863**
Closes: #25

### CODE REVIEW

File: src/main/java/com/dsrm/dsrmbackend/mappers/RoleMapper.java [hunk 1]
```diff
@@ -1,18 +1,24 @@
 package com.dsrm.dsrmbackend.mappers;
 
-import com.dsrm.dsrmbackend.entities.Role;
-import org.mapstruct.Mapper;
+        import com.dsrm.dsrmbackend.dto.RoleDTO;
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/mappers/RoleMapper.java]: Bad indent

File: src/main/java/com/dsrm/dsrmbackend/controllers/RoleController.java [hunk 2]
```diff
@@ -1,32 +1,55 @@
...
+import java.net.URI;
+import java.util.Optional;
 
 @RestController
 @RequiredArgsConstructor
 public class RoleController {
+    private final RoleService roleService;
+    private final RoleMapper roleMapper;
 
+    @Transactional
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/controllers/RoleController.java]: It is not necessary

File: src/main/java/com/dsrm/dsrmbackend/controllers/RoleController.java [hunk 3]
```diff
@@ -1,32 +1,55 @@
...
     @ResponseStatus(HttpStatus.CREATED)
     @PostMapping(value = "/roles", consumes = MediaType.APPLICATION_JSON_VALUE)
-    void addRole(@RequestBody RoleRequestDTO roleRequestDTO) {
+    public ResponseEntity<?> addRole(@RequestBody RoleRequestDTO roleRequestDTO) {
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/controllers/RoleController.java]: ```ResponseEntity<Void>```

File: src/main/java/com/dsrm/dsrmbackend/mappers/RoleMapper.java [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
+        import org.mapstruct.Mapping;
 
-import java.util.List;
-import java.util.Set;
+        import java.util.List;
+        import java.util.Set;
 
 @Mapper(componentModel = "spring")
 public interface RoleMapper {
+    @Mapping(target = "id", ignore = true)
```
Comment (HerbMat) [inline src/main/java/com/dsrm/dsrmbackend/mappers/RoleMapper.java]: WE don't need that annotation.

File: src/test/java/com/dsrm/dsrmbackend/RoomControllerTests.java [hunk 5]
```diff
@@ -22,6 +23,7 @@
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
 
 @Testcontainers
+@DirtiesContext
```
Comment (HerbMat) [inline src/test/java/com/dsrm/dsrmbackend/RoomControllerTests.java:26]: Why did you add this?

**Review by HerbMat — CHANGES_REQUESTED** [review]

File: src/test/java/com/dsrm/dsrmbackend/RoomControllerTests.java (hunk 5 above)
Comment (Boomber863) [reply src/test/java/com/dsrm/dsrmbackend/RoomControllerTests.java:26]: I couldn't run all the tests together without it, because TestContainers kills a container and creates a new one. SpringBootTest is not aware of that and tries to connect to an unavailable port

**Review by HerbMat — APPROVED** [review]
